homeassistant/components/airzone/climate.py

```diff
@@ -105,8 +115,42 @@ def __init__(
         self._attr_hvac_modes = [
             HVAC_MODE_LIB_TO_HASS[mode] for mode in self.get_airzone_value(AZD_MODES)
         ]
+        if (
+            self.get_airzone_value(AZD_SPEED) is not None
+            and self.get_airzone_value(AZD_SPEEDS) is not None
+        ):
+            self._set_fan_speeds()
+
         self._async_update_attrs()
 
+    def _set_fan_speeds(self) -> None:
+        self._attr_supported_features |= ClimateEntityFeature.FAN_MODE
+
+        speeds = self.get_airzone_value(AZD_SPEEDS)
+        max_speed = max(speeds)
+        if max_speed == 2:
+            self._speeds = {
+                0: FAN_AUTO,
+                1: FAN_LOW,
+                2: FAN_HIGH,
+            }
+        elif max_speed == 3:
+            self._speeds = {
```

Comment:
Does `self._speeds` need to be saved?  doesn't look used outside this function.

Reply:
Yes, it's used in `_async_update_attrs`:
```
if self.supported_features & ClimateEntityFeature.FAN_MODE:
            self._attr_fan_mode = self._speeds.get(self.get_airzone_value(AZD_SPEED))
```